backend/analytics_server/mhq/service/incidents/incidents.py

```diff
@@ -14,52 +19,159 @@
+    ) -> List[Incident]:
         incident_filter: IncidentFilter = apply_incident_filter(
             entity_type=EntityType.TEAM,
             entity_id=team_id,
             setting_types=[
                 SettingType.INCIDENT_SETTING,
                 SettingType.INCIDENT_TYPES_SETTING,
+                SettingType.INCIDENT_PRS_SETTING,
             ],
         )
-        return self._incidents_repo_service.get_team_incidents(
+        incidents = self._incidents_repo_service.get_team_incidents(
             team_id, interval, incident_filter
         )
+        pr_incidents: List[Incident] = self.get_team_pr_incidents(
+            team_id, interval, pr_filter
+        )
+
+        total_incidents = incidents + pr_incidents
+        total_incidents = sorted(total_incidents, key=lambda x: x.creation_date)
+
+        return list({incident.key: incident for incident in total_incidents}.values())
+
+    def get_team_pr_incidents(
+        self, team_id: str, interval: Interval, pr_filter: PRFilter
+    ) -> List[Incident]:
+
+        settings = self._settings_service.get_settings(
+            setting_type=SettingType.INCIDENT_PRS_SETTING,
+            entity_type=EntityType.TEAM,
+            entity_id=team_id,
+        )
+
+        if not settings:
+            return []
+
+        incident_prs_setting: IncidentPRsSetting = settings.specific_settings
+
+        if not incident_prs_setting.filters:
+            return []
+
+        team_repo_ids = list(
+            tr.org_repo_id
+            for tr in self._code_repo_service.get_active_team_repos_by_team_id(team_id)
+        )
+        resolution_prs_interval = Interval(
+            from_time=interval.from_time, to_time=time_now()
+        )
+        resolution_prs_filter: PRFilter = apply_pr_filter(
+            asdict(pr_filter),
+            EntityType.TEAM,
+            team_id,
+            [SettingType.EXCLUDED_PRS_SETTING, SettingType.INCIDENT_PRS_SETTING],
+        )
+
+        resolution_prs = self._code_repo_service.get_prs_merged_in_interval(
+            team_repo_ids, resolution_prs_interval, resolution_prs_filter
+        )
+
+        pr_numbers: List[str] = []
+        pr_incidents: List[Incident] = []
+        repo_id_to_pr_number_to_pr_map: Dict[str, Dict[str, PullRequest]] = {}
+
+        for pr in resolution_prs:
+            for filter in incident_prs_setting.filters:
+                incident_pr_number = self._extract_pr_number_from_regex(
+                    getattr(pr, filter["field"]), filter["value"]
+                )
+
+                if incident_pr_number:
+                    pr_numbers.append(incident_pr_number)
+                    if str(pr.repo_id) not in repo_id_to_pr_number_to_pr_map:
+                        repo_id_to_pr_number_to_pr_map[str(pr.repo_id)] = {}
+                    repo_id_to_pr_number_to_pr_map[str(pr.repo_id)][
+                        incident_pr_number
+                    ] = pr
+                    break
```

Comment:
Can be made cleaner using defaultdicts(dict). Then you won't have to the extra conditional check to see `if str(pr.repo_id) not in repo_id_to_pr_number_to_pr_map`.